### DISCUSSION TITLE
How to configure a vitest.workspace.ts without having the root folder loaded (Workspace folder not found for file vitest.workspace.ts)

### DISCUSSION THREAD

**Discussion by WtfJoke** [vitest-dev/vscode#447] [Q&A]
I cant get a vitest workspace to work with our type of folder structure. Any help would be appreciated!

My project structure looks like that:
![image](https://github.com/user-attachments/assets/15c84bef-644c-4178-b0cd-abf21cb2a9c2)
All these projects can also be opened individually and are self contained (its no monorepo).

My vscode.workspace.ts looks like that:
```typescript
export default ["project1/vitest.config.ts"];
```

In my vs-code workspace `projects.code-workspace` Ive defined all projects like that:
```json
"folders": [
		{
			"path": "project1",
			"name": "My cool project 1"
		},
		{
			"path": "project2",
			"name": "My cool project 2"
		}
],
"settings": {
		"vitest.workspaceConfig": "~/projects/vitest.workspace.ts",
}
```

Now the vitest extensions crashes with 
```
[API] Using user workspace config: /home/manuel/projects/vitest.workspace.ts.
[Error Error] Workspace folder not found for file:///home/manuel/projects/vitest.workspace.ts. Does the file exist?
Error: Workspace folder not found for file:///home/manuel/projects/vitest.workspace.ts. Does the file exist?
```
If I add the root folder to vscode it seems to work.
Is there some way to make this work without adding the root folder to my vs code workspace?

**Comment by sheremet-va** [comment]
I am afraid that it's not possible to do that with the current extension architecture. You need a workspace folder for every config file. Feel free to create an issue, it should be possible to rewrite that.
